independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/model/AppArtifact.java

```diff
@@ -14,8 +14,11 @@
 /**
  * Represents an application (or its dependency) artifact.
  *
+ * @deprecated in favor of {@link ResolvedDependency} and {@link io.quarkus.maven.dependency.Dependency}.
+ *
  * @author Alexey Loubyansky
  */
+@Deprecated(forRemoval = true)
```

Comment:
Please add a `since = "3.11"` everywhere you added this annotation.

Reply:
Done